source/OutfitterPanel.cpp

```diff
@@ -330,427 +362,573 @@ double OutfitterPanel::DrawDetails(const Point &center)
+				price = player.StockDepreciation().Value(selectedOutfit, day, howManyPer);
 			}
 
-			const Outfit *ammo = selectedOutfit->AmmoStoredOrUsed();
-			if(ammo && ship->OutfitCount(ammo))
+			if(howManyPer)
 			{
-				// Determine how many of this ammo I must sell to also sell the launcher.
-				int mustSell = 0;
-				for(const pair<const char *, double> &it : ship->Attributes().Attributes())
-					if(it.second < 0.)
-						mustSell = max<int>(mustSell, it.second / ammo->Get(it.first));
+				// Buy as many as can be paid for and will fit.
+				player.Accounts().AddCredits(-price);
+				player.AddStock(selectedOutfit, -howManyPer);
 
-				if(mustSell)
+				// Put them into fleet cargo.
+				player.Cargo().Add(selectedOutfit, howManyPer);
+			}
+		}
+		// Note: Buying into storage not implemented. Why waste your money?
+	}
+	else if(fromLocation == OutfitLocation::Ship)
+	{
+		for(int i = howManyPer; i; i--)
+		{
+			// Get the ships that have the most of this outfit installed.
+			const vector<Ship *> shipsToOutfit = GetShipsToOutfit();
+			// Note: to get here, we have already confirmed that every ship in the selection
+			// has the outfit installed and that the outfit can be uninstalled in the first place.
+			if(shipsToOutfit.empty())
+				break;
+			for(Ship *ship : shipsToOutfit)
+			{
+				// Uninstall the outfit.
+				ship->AddOutfit(selectedOutfit, -1);
+				int required = selectedOutfit->Get("required crew");
+				if(required)
+					ship->AddCrew(-required);
+				// Adjust hired crew counts.
+				ship->Recharge();
+
+				// If the context is sale:
+				if(toLocation == OutfitLocation::Shop)
 				{
-					ship->AddOutfit(ammo, -mustSell);
-					if(toStorage)
-						mustSell -= storage.Add(ammo, mustSell);
-					if(mustSell)
+					// Do the sale.
+					int64_t price = player.FleetDepreciation().Value(selectedOutfit, day);
+					player.Accounts().AddCredits(price);
+					player.AddStock(selectedOutfit, 1);
+				}
+				// If the context is uninstalling, move the outfit into Storage
+				else if(toLocation == OutfitLocation::Storage)
+					// Move to storage.
+					player.Storage().Add(selectedOutfit, 1);
+				// Note: It would be easy to add conditional statements above to also support uninstall into cargo,
+				//		 this is not supported in the outfitter at this time.
+
+				// Move ammo to storage.
+				// Since some outfits have ammo, remove any ammo that must also be moved as there
+				// aren't enough supporting slots for said ammo once this outfit is removed.
+				const Outfit *ammo = selectedOutfit->AmmoStoredOrUsed();
+				if(ammo && ship->OutfitCount(ammo))
+				{
+					// Determine how many of this ammo we must uninstall to also uninstall the launcher.
+					int mustUninstall = 0;
+					for(const pair<const char *, double> &it : ship->Attributes().Attributes())
+						if(it.second < 0.)
+							mustUninstall = max<int>(mustUninstall, it.second / ammo->Get(it.first));
```

Comment:
Shouldn't this round up rather than do plain double to integer conversion?
```suggestion
							mustUninstall = max<int>(mustUninstall, ceil(it.second / ammo->Get(it.first)));
```